app/src/main/java/com/peluso/walletguru/model/Account.kt

```diff
@@ -1,27 +1,40 @@
 package com.peluso.walletguru.model
 
 import com.kirkbushman.araw.models.Submission
+import com.peluso.walletguru.model.Account.Companion.orderSubmissions
+import kotlin.math.abs
 
 /**
  * Class which represents a financial account to be tracked in our application. An account is linked
  * to certain subreddits, and based on the performance of the account these posts are prioritized
  */
-class Account(val type: AccountType, val currentBalance: Float) {
-
-    //TODO: make this able to be stored in a sqlite database or in sharedprefs so we can save a user's
-    // balance over time
+class Account(val type: AccountType, val currentBalance: Float, val percentageChange: Float) {
 
     companion object {
-        fun List<Account>.orderSubmissions(submissions: List<Submission>): List<Submission> {
-            //TODO: write the algorithm for how to order a list of submissions based on the
-            // performance of the account
-            return submissions
+        fun List<Account>.orderSubmissions(accountMap: Map<AccountType, List<Submission>>):
+                List<Submission> {
+            //TODO: refactor to use the new parameters (provide mapping of account type to submission list)
+            // I've kept your below code as a reference
+
+            val sortedAccounts = this.sortedByDescending { abs(it.percentageChange) }
+
+            val subredditRankings = sortedAccounts.map { it.type to sortedAccounts.indexOf(it) }.toMap()
+
+            val sortedAccountMap = ArrayList<Submission>();
+
+            subredditRankings.entries.forEach {
+                if (accountMap.containsKey(it.key)) {
+                    sortedAccountMap.addAll(accountMap.getValue(it.key))
```

Comment:
This is definitely an improvement, and it runs well on my device; however, I still think it's not the idea we had discussed. When you scroll down the list of posts, right now if you want to see anything Crypto for example, you need to scroll to the complete bottom of the list. I think we should stagger the posts by the order you create in `sortedAccounts`. So like if the ranking goes `MORTGAGE, SAVING, INVESTMENT, CRYPTO` I think the post order should be first a post from the mortgage category, then saving category, then investment, then crypto, then back to mortgage, and so on and so forth. If that doesn't make sense please feel free to ask clarifying questions.

Reply:
Just pushed changes for implementing it this way, please take another look. Seems to be working correctly for me.